pkg/workload/workload_generator/ddl_generator.go

```diff
@@ -0,0 +1,368 @@
+	m := tablePattern.FindStringSubmatch(ddl)
+	if m == nil {
+		return nil, errors.New("no table name")
+	}
+	// Extract and normalize the table name
+	tableName := m[1]
+	parts := strings.Split(tableName, ".")
+	for i := range parts {
+		parts[i] = strings.Trim(parts[i], `"`) // Remove quotes from parts
+	}
+	rawName := tableName                 // Save the original table name
+	tableName = strings.Join(parts, ".") // Reconstruct normalized table name
+
+	// Create a new TableSchema with the normalized name
+	ts := NewTableSchema(tableName, rawName)
+
+	// Extract the column definitions block from the DDL
+	bodyRe := regexp.MustCompile(`(?s)\((.*)\)\s*([^)]*)$`)
+	bodyMatch := bodyRe.FindStringSubmatch(ddl)
+	if bodyMatch == nil {
+		return nil, errors.New("no column block")
+	}
+	body := bodyMatch[1]
+
+	var partsList []string
+	buf := ""
+	depth := 0
+	for _, ch := range body {
+		switch ch {
+		case '(':
+			depth++
+			buf += string(ch)
+		case ')':
+			depth--
+			buf += string(ch)
+		case ',':
+			if depth == 0 {
+				partsList = append(partsList, strings.TrimSpace(buf))
+				buf = ""
+			} else {
+				buf += string(ch)
+			}
+		default:
+			buf += string(ch)
+		}
+	}
+	if strings.TrimSpace(buf) != "" {
+		partsList = append(partsList, strings.TrimSpace(buf))
+	}
+	// Separate column definitions from table-level constraints
+	var colDefs, tableConstraints []string
+	for _, p := range partsList {
+		up := strings.ToUpper(strings.TrimSpace(p))
+		if strings.HasPrefix(up, "CONSTRAINT") || strings.HasPrefix(up, "PRIMARY KEY") || strings.HasPrefix(up, "UNIQUE") || strings.HasPrefix(up, "FOREIGN KEY") || strings.HasPrefix(up, "CHECK") || strings.HasPrefix(up, "INDEX") {
+			tableConstraints = append(tableConstraints, p)
+		} else {
+			colDefs = append(colDefs, p)
+		}
+	}
+	// Define regex pattern to extract column properties from column definitions
+	colPattern := regexp.MustCompile(`(?i)^\s*("?[^"]+"|[\w-]+)"?\s+([^\s]+)(?:\s+(NOT\s+NULL|NULL))?(?:\s+DEFAULT\s+((?:\([^\)]*\)|[^\s,]+)))?(?:\s+PRIMARY\s+KEY)?(?:\s+UNIQUE)?(?:\s+REFERENCES\s+([\w\.]+)\s*\(\s*([\w]+)\s*\))?(?:\s+CHECK\s*\(\s*(.*?)\s*\))?`)
+	inlinePKCols := []string{}
+	// Process each column definition
+	for _, cd := range colDefs {
+		m := colPattern.FindStringSubmatch(cd)
+		if m == nil {
+			continue // Skip if pattern doesn't match
+		}
+		// Extract column properties from regex matches
+		name := m[1]             // Column name
+		columnType := m[2]       // Column type
+		nullSpec := m[3]         // NULL or NOT NULL specification
+		defaultVal := m[4]       // DEFAULT value
+		foreignKeyTable := m[5]  // Referenced table for foreign keys
+		foreignKeyColumn := m[6] // Referenced column for foreign keys
+
+		ts.ColumnOrder = append(ts.ColumnOrder, name)
+		// Extract CHECK constraint if present (requires special handling for nested parentheses)
+		inlineCheck := ""
+		checkIdx := regexp.MustCompile(`(?i)\bCHECK\s*\(`).FindStringIndex(cd)
```

Comment:
Please note that regexp.MustCompile is relatively costly compared to reusing a compiled regex. Here's why and how you should decide whether to make it a var even if used only once.